### PR TITLE
Media build patches for 3.14

### PR DESCRIPTION
**Pull request by kszaq**
This is a set of patches that makes it possible to build amlvideo driver as a module and a part of media_build tree. It doesn't break our current usage of amlvideo.

### CODE REVIEW

File: drivers/amlogic/video_dev/Kconfig [hunk 1]
```diff
@@ -14,6 +15,7 @@ config V4L_AMLOGIC_VIDEO2
 	select VIDEO_DEV
 	select VIDEOBUF_GEN
 	select VIDEO_V4L2_COMMON
+	select VIDEOBUF_GEN
```
Comment (Ray-future) [inline drivers/amlogic/video_dev/Kconfig]: Is there a reason why VIDEOBUF_GEN is selected twice?

Comment (kszaq) [reply drivers/amlogic/video_dev/Kconfig]: One for `V4L_AMLOGIC_VIDEO` and one for `V4L_AMLOGIC_VIDEO2`. Each can be built separately and both needs VIDEOBUF_GEN.

Comment (Ray-future) [reply drivers/amlogic/video_dev/Kconfig]: Yes of course but inV4L_ AMLOGIC_VIDEO2 you have it twice.

Comment (kszaq) [reply drivers/amlogic/video_dev/Kconfig]: Ah yes, sure. Will fix.

**Comment by kszaq** [comment]
Fixed double selection and rebased. Thank you for your review @Raybuntu !

**Review by Ray-future — APPROVED** [review]

**Review by codesnake — COMMENTED** [review]
Please don't merge, this will need some changes.

**Comment by codesnake** [comment]
Generally I'm OK with this PR. One note is CONFIG_RATIONAL patch, perhaps it should be selected only when CONFIG_VIDEO_V4L2 is selected as it is done upstream: https://github.com/torvalds/linux/blob/master/drivers/media/v4l2-core/Kconfig#L9

**Comment by codesnake** [comment]
@kszaq This is good to go. Just remove the 3b5219b commit please. It is already a part of already merged PR https://github.com/LibreELEC/linux-amlogic/pull/58.

**Comment by kszaq** [comment]
Rebased and added regmap-related patches cherry-picked from mainline kernel to support latest media_build.

**Pull request merged**